### ISSUE TITLE
Best practices / example results and run settings for Dreambooth inpainting  

### ISSUE THREAD

**Issue by aaronlohwy** [huggingface/diffusers#1678]
Hi @thedarkzeno  @patil-suraj 

I'm trying to run the script here, https://github.com/huggingface/diffusers/tree/main/examples/research_projects/dreambooth_inpaint
(ty for this)

with similar setup as with training regular text=>image dreambooth. I'm finding that the quality of the generations for inpainting is far inferior to those that I get from pure text => image generation. 

Wondering if there are any rules of thumbs that I should be following in order to get better results (e.g. does it require longer training, should I be modifying the masking method during training etc.) Are there any sample runs/results from this pipeline that I can reference?

Thanks!

**Comment by patrickvonplaten** [comment]
Interesting! I don't know really and sadly won't have the time to look into it :sweat_smile: @thedarkzeno maybe?

**Comment by thedarkzeno** [comment]
The model uses random masks, this may cause it to learn a bit slower, for me it worked well with more steps like 500-1000, but it could be be different for your dataset. I'm Thinking about adding some segmentation option to optimize the training, but not sure if I will have time soon.

**Comment by aaronlohwy** [comment]
Thanks for the clarification - I've been training for up to 1k steps on a concept with ~7 images. I think the main issue is that it doesn't seem to be learning the concept - I tested this by basically providing an all-white mask on top of a blank base, and it was unable to re-generate the concept (though typically dreambooth would be able to overfit very quickly). Are there any results / reference runs or notebooks? I can also play around with hparams but wanted to get an e2e run working first.

As a side note - if there's a general inpainting training script, (similar to diffusers/examples/text_to_image/train_text_to_image.py)? I could also try that if so.

**Comment by belonel** [comment]
I had the same issue with the inpainting script
(https://github.com/huggingface/diffusers/tree/main/examples/research_projects/dreambooth_inpaint).

It didn't learn the concept on 100-500 training steps, using 10 example images. 

This helped for me: 
- training with my own masks;
- training text encoder as it was suggested in the readme. 
- increasing training steps to 2000

Here are the results for a sweatshirt. 
![2023-01-08 21 36 32](https://user-images.githubusercontent.com/25452700/211212960-61851ead-59d8-4aca-aee0-2c9ca9742fcb.jpg)

Still, strange that inpainting takes much more training steps than a dreambooth text2image. @thedarkzeno Do you have any suggestions on why this might be the case?

**Comment by thedarkzeno** [comment]
Hey @belonel could you elaborate on the masks that you used?
I don't know what masks you used, but I believe the training for inpainting takes longer because of the masks. I think that using a model like clipseg to generate masks could make it better, but I haven't had time to implement it.

**Comment by belonel** [comment]
Thanks, I'll take a look at the clipseg. I used [cloth segmentation using u2net](https://github.com/levindabhi/cloth-segmentation) to train the model to reproduce clothes. Masks look like this: 
![2](https://user-images.githubusercontent.com/25452700/211480657-12d6f673-9be9-4acf-a0f0-9b3201b94eb9.png)
![1](https://user-images.githubusercontent.com/25452700/211480682-0a641eb1-7d4a-4992-a9bd-a111c5231aaa.png)

**Comment by thedarkzeno** [comment]
@belonel these masks look pretty good, they must be much better for training the model compared to the random ones.

**Comment by kadirnar** [comment]
@belonel ,
Do you have a sample notebook? How can I do that?
